app/controllers/goals_controller.rb

```diff
@@ -0,0 +1,98 @@
+class GoalsController < ApplicationController
+
+  def index
+    @messages = Message.broadcast
+    @sorting_method = params[:sort]
+    if @sorting_method == "Newest"
+      @goals = Goal.sort_time
+    elsif @sorting_method == "Submitted"
+      @goals = Goal.sort_setter
+    elsif @sorting_method == "Charity"
+      @goals = Goal.sort_charity
+    end
+  end
+
+  def new
+    @messages = Message.broadcast
+    @goal = Goal.new
+    @charities = Charity.all
+    @milestone = Milestone.new
+  end
+
+  def show
+    @goal = Goal.includes(:setter, :tender, :milestones, :messages, :pledges).find_by(id: params[:id])
+    @milestones = @goal.milestones.order(:deadline)
+    @message = Message.new
+    @messages = @goal.messages.order(created_at: :desc)
+    @errors = params[:errors]
+  end
+
+  def create
+    @messages = Message.broadcast
+    @goal = Goal.new(goal_params)
+    @goal.setter = current_user
+    @goal.charity = Charity.find_by(name: params[:charity_selector])
+    @goal.tender = User.find_by(username: params[:goal][:tender])
+    @mile_array = make_milestones(@goal)
+    @mile_array.each do |milestone|
+      if milestone.deadline && milestone.description && @goal.save
```

Comment:
You shouldn't need to do this save followed by setting goal_id here. If you use @goal.milestones.build on line 93, this method can do just @goal.save and AR will save all of the records.